aptos-move/aptos-vm-types/src/resource_group_adapter.rs

```diff
@@ -168,17 +170,23 @@ impl TResourceGroupView for ResourceGroupAdapter<'_> {
     fn release_group_cache(
         &self,
     ) -> Option<HashMap<Self::GroupKey, BTreeMap<Self::ResourceTag, Bytes>>> {
-        // TODO: once block executor can handle the new VMChangeSet, return None when
-        // group_size_kind is GroupSizeKind::AsSum, which will also switch output of session
-        // to a newer format and correspondingly, to the new (AsSum) gas charging for it.
-        // Note: should still flush the cache, even if returning None.
-        Some(
-            self.group_cache
-                .borrow_mut()
-                .drain()
-                .map(|(k, v)| (k, v.0))
-                .collect(),
-        )
+        if self.group_size_kind == GroupSizeKind::AsSum {
```

Comment:
2/2 we need to feature gate here, i.e. if not enabled don't go into this branch (return some).

For capabilities, when not capable, we will return an indicator here that will force the VM to produce V0 resource group change set but charge as V1 (I'm thinking to turn V0/V1 enum into struct there and prepare both, shouldn't be too complex and well-testable) - we can do that next as one of the preconditions for enabling group feature flag -> aggr2 feature flag.

Reply:
@igor-aptos